internal/cluster/cluster_test.go

```diff
@@ -1491,6 +1543,154 @@ var _ = Describe("SetVips", func() {
 	})
 })
 
+var _ = Describe("Majority groups", func() {
+	var (
+		dbIndex    int
+		clusterApi *Manager
+		db         *gorm.DB
+		id         strfmt.UUID
+		cluster    common.Cluster
+		ctrl       *gomock.Controller
+		mockEvents *events.MockHandler
+		dbName     = "cluster_majority_groups"
+	)
+
+	AfterEach(func() {
+		common.DeleteTestDB(db, dbName)
+	})
+
+	BeforeEach(func() {
+		db = common.PrepareTestDB(dbName, &events.Event{})
+		dbIndex++
+		ctrl = gomock.NewController(GinkgoT())
+		mockEvents = events.NewMockHandler(ctrl)
+		dummy := &leader.DummyElector{}
+		clusterApi = NewManager(getDefaultConfig(), getTestLog().WithField("pkg", "cluster-monitor"), db,
+			mockEvents, nil, nil, dummy)
+
+		id = strfmt.UUID(uuid.New().String())
+		cluster = common.Cluster{Cluster: models.Cluster{
+			ID:                       &id,
+			Status:                   swag.String(models.ClusterStatusReady),
+			MachineNetworkCidr:       "1.2.3.0/24",
+			BaseDNSDomain:            "test.com",
+			PullSecretSet:            true,
+			ServiceNetworkCidr:       "1.2.4.0/24",
+			ClusterNetworkCidr:       "1.3.0.0/16",
+			ClusterNetworkHostPrefix: 24,
+		}}
+		Expect(db.Create(&cluster).Error).ShouldNot(HaveOccurred())
+	})
+	Context("Set Majority groups", func() {
+		It("Empty", func() {
+			addInstallationRequirementsWithConnectivity(id, db)
+
+			cluster = geCluster(*cluster.ID, db)
+			Expect(swag.StringValue(cluster.Status)).Should(Equal(models.ClusterStatusReady))
+			Expect(len(cluster.Hosts)).Should(Equal(3))
+			Expect(clusterApi.SetConnectivityMajorityGroupsForCluster(&cluster, db)).ToNot(HaveOccurred())
+			c := getCluster(*cluster.ID, db)
```

Comment:
There is a lot of common code in the tests consider creating a helper function.

Reply:
Will try